homeassistant/config_entries.py

```diff
@@ -3485,7 +3485,25 @@ async def async_finish_flow(
         entry = self.hass.config_entries.async_get_known_entry(flow.handler)
 
         if result["data"] is not None:
-            self.hass.config_entries.async_update_entry(entry, options=result["data"])
+            automatic_reload = False
+            if (
+                hasattr(flow, "automatic_reload") is True
+                and flow.automatic_reload is True  # type: ignore[attr-defined]
+            ):
+                automatic_reload = True
```

Comment:
I think this is cleaner
```suggestion
            automatic_reload = False
            if isinstance(flow, OptionsFlowWithReload):
                automatic_reload = flow.automatic_reload
```